torch/_inductor/fx_passes/micro_pipeline_tp.py

```diff
@@ -211,8 +211,25 @@ class _ReduceScatterMatch:
     group_name: str
 
     def replace_with(self, new_node: torch.fx.Node) -> None:
+        # Replace all uses of the result node (wait_tensor) with the fused node.
         self.res_node.replace_all_uses_with(new_node)
```

Comment:
so this node is presumably replacing all usages of the original `reduce_scatter` output node with the new fused node. Does that sound right? If that sounds right, maybe we should look into why it is not successfully replacing the usage by the `target == 'output'` node in the graph?

Reply:
This line is actually replacing all usages of the result node (wait_tensor) with the new fused node. IMO these variable names are a bit ambiguous (`rs_node` vs `res_node`) so I just pushed a change changing:

- `rs_node` => `reduce_scatter_node`
- `res_node` => `wait_tensor_node` (for ReduceScatterMatch, where the result node is explicitly always a wait_tensor).

Hopefully that helps clarify the approach here. Basically we are replacing all uses of the result/wait_tensor with the fused node, and then if the reduce_scatter node is being used by `output` (i.e., saved for backward), we change output to use the fused node instead.

Reply:
makes sense, thanks for explaining!

Let me know if you need me to review the PR. Happy to review it - also @kwen2501 might be interested since it touches the asyncTP code.

Reply:
Ok awesome! Yes if you and @kwen2501 want to review that would be greatly appreciated, thanks